mesonbuild/modules/python.py

```diff
@@ -57,11 +57,17 @@ class FindInstallationKw(ExtractRequired):
         modules: T.List[str]
         pure: T.Optional[bool]
 
+    class ExtensionModuleKw(SharedModuleKw):
+
+        pass
+
 
 mod_kwargs = {'subdir'}
 mod_kwargs.update(known_shmod_kwargs)
 mod_kwargs -= {'name_prefix', 'name_suffix'}
 
+_MOD_KWARGS = [k for k in SHARED_MOD_KWS if k.name not in {'name_prefix', 'name_suffix'}]
```

Comment:
`mod_kwargs` looks like it could be derived from `SHARED_MOD_KWARGS`. The names being so similar is also confusing

Reply:
No, it can't, because `SHARED_MOD_KWS` isn't complete yet. If it was we wouldn't need `mod_kwargs` at all, since it's only used for `permittedKwargs`, which is a predecessor for `typed_kwargs`